### PR TITLE
added scheduling and qos docs

### PR DESCRIPTION
**Pull request by puja108**
RFR @marians @oponder @webwurst

### CODE REVIEW

File: content/guides/scheduling-constraints-and-resource-qos/index.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
++++
+title = "Scheduling Constraints and Resource Quality of Service"
+description = "How you can constraint resource usage and define Quality of Service classes in Kubernetes"
```
Comment (oponder) [inline content/guides/scheduling-constraints-and-resource-qos/index.md]: constraint -> constrain

File: content/guides/scheduling-constraints-and-resource-qos/index.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+## Defining Resource Constraints for Pods
+
+You can specify requests and limits for CPU and RAM usage for each container of a Pod by setting:
+
+- `spec.container[].resources.limits.cpu`
+- `spec.container[].resources.limits.memory`
+- `spec.container[].resources.requests.cpu`
+- `spec.container[].resources.requests.memory.`
+
+If request values are not set, they defaut to the value set as limit. Note that limits must always be equal to or greater than requests.
```
Comment (oponder) [inline content/guides/scheduling-constraints-and-resource-qos/index.md]: defaut => default

File: content/guides/scheduling-constraints-and-resource-qos/index.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+Note that even when actual memory or CPU usage might be low, the scheduler will still refuse to place pods onto nodes if the capacity check fails. This protects against a resource shortage when resource usage increases, e.g. due to a daily peak or other kind of load increase.
+
+## Resource Guarantees
+
+Kubernetes differentiates between compressible and incompressible resources. Former currently supports CPU and latter currently only Memory.
+
+For compressible resources, Pods are guaranteed to get the amount they requested, they might or might not get additional resources. If Pods exceed their limits they will be throttled accordingly. If no limit is set they may use as much of the compressible resource as available.
+
+For incompressible resources, Pods are guaranteed to get the amount they requested. If they exceed their request they might get killed (e.g. when another pod requests more of the resource). if Pods use more than their limit, the process that is using the most amount of memory, inside one of the pod's containers, will be killed by the kernel.
```
Comment (marians) [inline content/guides/scheduling-constraints-and-resource-qos/index.md]: `if Pods` -> `If Pods`

File: content/guides/scheduling-constraints-and-resource-qos/index.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+The requests and limits of a Pod are defined by the sum of the respective values of all its containers (unset values are treated as zero depending on your cluster configuration).
+
+### CPU
+
+CPU resoures are measured in (v)Core equivalents. You can specify it in decimals (e.g. `0.5` meaning half a core) or in `milicpu` (e.g. `500m` meaning half a core).
+
+### Memory
+
+Memory resources are measured in bytes. You specify them as plain integer or as fixed-point integers with one of SI suffixes (E, P, T, G, M, K) or their power-of-two equivalents (Ei, Pi, Ti, Gi, Mi, Ki). For example, the following represent roughly the same value: 128974848, 129e6, 129M, 123Mi.
```
Comment (puja108) [inline content/guides/scheduling-constraints-and-resource-qos/index.md]: yeah, it's the same, will change anyway, this is too complex

File: content/guides/scheduling-constraints-and-resource-qos/index.md (hunk 3 above)
Comment (puja108) [inline content/guides/scheduling-constraints-and-resource-qos/index.md]: done

**Pull request merged**